### ISSUE TITLE
Move instance configuration to `deploy`

### ISSUE THREAD

**Issue by pawiecz** [kernelci/kernelci-pipeline#470]
1)We have production configs included in kernelci-core AND api AND pipeline
2)On staging we apply patches stored in kernelci-deploy, to convert it to "staging" configs

Every time we update production more or less significantly, it breaks those patches.
So for long term, **i think we need to open issue about config storage/management**. IMHO staging, production configs should be stored in deploy in separate directories, and software distributions, like you said, have minimal working config, friendly for those who want to try software.

_Originally posted by @nuclearcat in https://github.com/kernelci/kernelci-pipeline/pull/440#discussion_r1497036328_

**Comment by nuclearcat** [comment]
Absolutely, as it also become hell to maintain config files for kubernetes deployment as well.
